src/backend/distributed/operations/shard_transfer.c

```diff
@@ -376,14 +376,33 @@ citus_move_shard_placement(PG_FUNCTION_ARGS)
 		PlacementMovedUsingLogicalReplicationInTX = true;
 	}
 
+	DropOrphanedResourcesInSeparateTransaction();
```

Comment:
This seems like a behavioural change. I'm kind of wondering whether we really want this (e.g. what if this fails on some unrelated node?), and whether we want it here (e.g. if we have parallel moves, should this be in rebalancer_start?).

Reply:
Does it have a downside to have it here? We take locks on operationIds during the cleanup anyway. Besides, the commands we send to the workers to drop the objects are optional. What could be the problem in case it fails to drop some object?

Reply:
I think this is indeed fine to do since the commands that are run are optional. It seems desirable, to allow for back and forth movement of shards.

Reply:
Also, we already did this sometimes in `EnsureShardCanBeCopied` (and it's removed there now).